nuxeo-webdav-test/src/test/java/org/nuxeo/ecm/webdav/WebDavServerFeature.java

```diff
@@ -50,60 +57,86 @@
 
-        ServletAdapter jerseyAdapter = new ServletAdapter();
         Application app = new org.nuxeo.ecm.webdav.Application();
         ApplicationAdapter conf = new ApplicationAdapter(app);
         conf.getFeatures().put(ResourceConfig.FEATURE_MATCH_MATRIX_PARAMS, Boolean.TRUE);
-        jerseyAdapter.setServletInstance(new ServletContainer(conf));
-        jerseyAdapter.addRootFolder(path);
-        jerseyAdapter.setHandleStaticResources(true);
-        jerseyAdapter.setContextPath("");
-        // session cleanup
-        jerseyAdapter.addFilter(new RequestContextFilter(), "RequestContextFilter", null);
-        jerseyAdapter.addFilter(new SessionCleanupFilter(), "SessionCleanupFilter", null);
-        jerseyAdapter.addFilter(new NuxeoAuthenticationFilter(), "NuxeoAuthenticationFilter", null);
-        jerseyAdapter.addFilter(new WebEngineFilter(), "WebEngineFilter", null);
-
-        if (DEBUG) {
-            jerseyAdapter.addInitParameter("com.sun.jersey.spi.container.ContainerRequestFilters",
-                    "com.sun.jersey.api.container.filter.LoggingFilter");
-            jerseyAdapter.addInitParameter("com.sun.jersey.spi.container.ContainerResponseFilters",
-                    "com.sun.jersey.api.container.filter.LoggingFilter");
-        }
-        jerseyAdapter.addInitParameter("com.sun.jersey.config.feature.logging.DisableEntitylogging", "true");
+        String servletName = "testServlet";
+        Servlet servlet = new ServletContainer(conf);
+        tomcat.addServlet(CONTEXT, servletName, servlet);
+        context.addServletMappingDecoded("/*", servletName);
+        addFilter(context, servletName, "RequestContextFilter", new RequestContextFilter());
+        addFilter(context, servletName, "SessionCleanupFilter", new SessionCleanupFilter());
+        addFilter(context, servletName, "NuxeoAuthenticationFilter", new NuxeoAuthenticationFilter());
+        addFilter(context, servletName, "WebEngineFilter", new WebEngineFilter());
+
+        tomcat.start();
+    }
 
-        gws.addGrizzlyAdapter(jerseyAdapter, new String[] { "" });
+    protected void addFilter(Context context, String servletName, String filterName, Filter filter) {
+        FilterDef filterDef = new FilterDef();
+        filterDef.setFilterName(filterName);
+        filterDef.setFilterClass(filter.getClass().getName());
+        context.addFilterDef(filterDef);
+        FilterMap filterMap = new FilterMap();
+        filterMap.setFilterName(filterName);
+        filterMap.addServletName(servletName);
+        context.addFilterMap(filterMap);
+    }
 
-        // let Grizzly run
-        try {
-            gws.start();
-        } catch (IOException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
+    protected void tearDownTomcat() throws Exception {
+        if (tomcat != null) {
+            tomcat.stop();
+            tomcat.destroy();
+            Thread.sleep(100);
```

Comment:
Why the Thread.sleep?

Reply:
Copied from existing code in `CmisFeatureSessionHttp`.